src/coreclr/src/jit/morph.cpp

```diff
@@ -2890,9 +2890,65 @@ void Compiler::fgInitArgInfo(GenTreeCall* call)
     SYSTEMV_AMD64_CORINFO_STRUCT_REG_PASSING_DESCRIPTOR structDesc;
 #endif // UNIX_AMD64_ABI
 
+#if defined(DEBUG)
+    // Check that we have valid inforamtion about call's argument types.
+    // For example:
+    // load byte; call(int) -> CALL(PUTARG_TYPE byte(IND byte));
+    // load int; call(byte) -> CALL(PUTARG_TYPE int (IND int));
+    // etc.
+    if ((call->callSig != nullptr) && (Target::g_tgtArgOrder == Target::ARG_ORDER_R2L))
```

Comment:
`callSig == nullptr` means mostly a helper call, all helper calls that I have checked are working with i4 and i8 types but I expect I will have to add more `PUTARG_TYPE` there once I run more tests on arm64 OSX and find bugs with missed type information.